src/bp-members/classes/class-bp-members-notices-rest-controller.php

```diff
@@ -0,0 +1,1090 @@
+		 *
+		 * @param bool|WP_Error   $retval  Returned value.
+		 * @param WP_REST_Request $request The request sent to the API.
+		 */
+		return apply_filters( 'bp_members_notices_rest_update_item_permissions_check', $retval, $request );
+	}
+
+	/**
+	 * Dismisses the currently active notice for the current user.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function dismiss_notice( $request ) {
+		$notice_id = $request->get_param( 'id' );
+
+		// Mark the requested notice as closed.
+		if ( $notice_id ) {
+			$notice = bp_members_get_notice( $notice_id );
+
+			// Mark the first priority notice as closed.
+		} else {
+			$notice = bp_get_active_notice_for_user();
+		}
+
+		if ( is_null( $notice ) ) {
+			return new WP_Error(
+				'bp_rest_invalid_id',
+				__( 'Sorry, this notice does not exist.', 'buddypress' ),
+				array(
+					'status' => 404,
+				)
+			);
+		}
+
+		// Get Previous active notice.
+		$previous = $this->prepare_item_for_response( $notice, $request );
+
+		// Dismiss the active notice for the current user.
+		$dismissed = bp_members_dismiss_notice( bp_loggedin_user_id(), $notice->id );
+
+		// Build the response.
+		$response = new WP_REST_Response();
+		$response->set_data(
+			array(
+				'dismissed' => $dismissed,
+				'previous'  => $previous->get_data(),
+			)
+		);
+
+		/**
+		 * Fires after a notice is dismissed via the REST API.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param BP_Members_Notice $notice   Notice object.
+		 * @param WP_REST_Response  $response The response data.
+		 * @param WP_REST_Request   $request  The request sent to the API.
+		 */
+		do_action( 'bp_members_notices_rest_dismiss_notice', $notice, $response, $request );
+
+		return $response;
+	}
+
+	/**
+	 * Check if a given request has access to dismiss the current notice.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool
+	 */
+	public function dismiss_notice_permissions_check( $request ) {
+		$retval = new WP_Error(
+			'bp_rest_authorization_required',
+			__( 'Sorry, you are not allowed to dismiss notices.', 'buddypress' ),
+			array(
+				'status' => rest_authorization_required_code(),
+			)
+		);
+
+		if ( is_user_logged_in() ) {
+			$retval = true;
+		}
+
+		/**
+		 * Filter the notices `dismiss_notice` permissions check.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param bool|WP_Error   $retval  Returned value.
+		 * @param WP_REST_Request $request The request sent to the API.
+		 */
+		return apply_filters( 'bp_members_notices_rest_dismiss_notice_permissions_check', $retval, $request );
+	}
+
+	/**
+	 * Delete a notice.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function delete_item( $request ) {
+		// Setting context.
+		$request->set_param( 'context', 'edit' );
```

Comment:
See my feedback about setting this manually.

Reply:
Yes I have the same question: any idea why you set it this way in the first place?
See https://github.com/buddypress/BP-REST/blob/master/includes/bp-messages/classes/class-bp-rest-sitewide-notices-endpoint.php#L541